core-bundle/src/EventListener/ClearSessionDataListener.php

```diff
@@ -31,6 +40,11 @@ public function __invoke(ResponseEvent $event): void
             return;
         }
 
+        if ($this->scopeMatcher->isContaoMainRequest($event)) {
+            // Disable the default Symfony auto cache control
+            $event->getResponse()->headers->set(AbstractSessionListener::NO_AUTO_CACHE_CONTROL_HEADER, '1');
+        }
+
```

Comment:
Instead of extracting setting `NO_AUTO_CACHE_CONTROL_HEADER` from the `MakeResponsePrivateListener` it would be more consistent to keep it there - but transform the `MakeResponsePrivateListener` to an event subscriber with two methods: one that sets `NO_AUTO_CACHE_CONTROL_HEADER` with a priority _before_ the `SessionListener` and the other method that does the rest (i.e. sets the response to private if applicable).

Reply:
Done in 96c8f5b0225744dd9a1f4fb1229f22ea2583354d